spec/moonshot/plugins/asg_spec.rb

```diff
@@ -108,64 +118,35 @@ def stub_cf_client
     end
   end
 
-  describe '#initialize' do
-    before(:each) do
-      allow(System).to receive(:new)
-        .and_return(system)
-      stub_cf_client
-    end
-    context 'when MOONSHOT_SSH_USER is not defined' do
-      it 'uses LOGNAME for ssh_user' do
-        ENV['LOGNAME'] = 'SemiCoolDude'
-        ENV.delete('MOONSHOT_SSH_USER')
-        expect(subject.instance_variable_get(:@ssh_user))
-          .to eq('SemiCoolDude')
-      end
-    end
-
-    context 'when MOONSHOT_SSH_USER is defined' do
-      it 'uses MOONSHOT_SSH_USER for ssh_user' do
-        ENV['MOONSHOT_SSH_USER'] = 'CoolDude'
-        expect(subject.instance_variable_get(:@ssh_user))
-          .to eq('CoolDude')
-      end
-    end
-  end
-
   describe '#shutdown_instance' do
-    let(:hostname) { 'ec2-54-236-102-14.compute-1.amazonaws.com' }
+    let(:public_ip_address) { '10.234.32.21' }
     let(:instance) { instance_double(Aws::EC2::Instance) }
+    let(:command_builder) { Moonshot::SSHCommandBuilder }
     subject { super().send(:shutdown_instance, instance_id) }
 
+
     before(:each) do
+      moonshot_config.ssh_config.ssh_user = 'ci_user'
       allow(Aws::EC2::Instance).to receive(:new).and_return(instance)
-      allow(instance).to receive(:public_dns_name) \
-        .and_return(hostname)
-      allow(System).to receive(:exec)
-    end
-
-    it 'looks up the DNS name of the host' do
-      expect(instance).to receive(:public_dns_name) \
-        .and_return(hostname)
-      subject
+      allow_any_instance_of(command_builder).to receive(:instance_ip).and_return(public_ip_address)
+      allow(instance).to receive(:wait_until_stopped)
     end
 
-    it 'issues a shutdown to the instance' do
-      ENV['MOONSHOT_SSH_USER'] = 'ci_user'
-      expect(System).to receive(:exec).with(
-        /ssh (.*) ci_user@#{hostname} 'sudo shutdown -h now'/,
-        raise_on_failure: false
+    it 'issues a shutdown without options to the instance' do
+      moonshot_config.ssh_config.ssh_options = nil
+      expect_any_instance_of(ssh_executor).to receive(:run).with(
+        "ssh -t -l #{moonshot_config.ssh_config.ssh_user} #{public_ip_address} sudo\\ shutdown\\ -h\\ now"
       )
       subject
     end
 
-    it 'runs SSH with proper option to ignore host keys' do
-      ENV['MOONSHOT_SSH_USER'] = 'ci_user'
-      opts_string = '-o UserKnownHostsFile=/dev/null ' \
-                    '-o StrictHostKeyChecking=no'
-      expect(System).to receive(:exec).with(
-        /#{opts_string}/,
-        any_args
+    it 'issues a shutdown with options to the instance' do
+      moonshot_config.ssh_config.ssh_options =
+        '-v -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o ConnectTimeout=10'
```

Comment:
We don't have to provide all these options here. Just provide "-v  -o userknownhostsfile"